<?xml version="1.0"?>
<launch>
  <!-- Load camera parameters -->
  <rosparam command="load" file="$(find dual_camera_simulation)/config/camera1_params.yaml" />
  <rosparam command="load" file="$(find dual_camera_simulation)/config/camera2_params.yaml" />

  <!-- Camera 1 processing nodes -->
  <group ns="camera1">
    <!-- Rectify the images -->
    <node name="image_proc" pkg="image_proc" type="image_proc"/>
    
    <!-- Run point cloud filters -->
    <node name="point_cloud_filter" pkg="dual_camera_simulation" type="point_cloud_filter_node" output="screen">
      <param name="input_cloud" value="/camera1/depth/points" />
      <param name="output_cloud" value="/camera1/depth/filtered_points" />
      <param name="camera_name" value="camera1" />
    </node>
  </group>

  <!-- Camera 2 processing nodes -->
  <group ns="camera2">
    <!-- Rectify the images -->
    <node name="image_proc" pkg="image_proc" type="image_proc"/>
    
    <!-- Run point cloud filters -->
    <node name="point_cloud_filter" pkg="dual_camera_simulation" type="point_cloud_filter_node" output="screen">
      <param name="input_cloud" value="/camera2/depth/points" />
      <param name="output_cloud" value="/camera2/depth/filtered_points" />
      <param name="camera_name" value="camera2" />
    </node>
  </group>
</launch>